src/components/encoder.ts

```diff
@@ -0,0 +1,2 @@
+export type { Encoder } from './encoder/Encoder';
```

Comment:
if you agree with re-exporting the type

```suggestion
export type { Encoder, EncoderProperties } from './encoder/Encoder';
```